sqlglot/dialects/hive.py

```diff
@@ -51,6 +51,17 @@
 DIFF_MONTH_SWITCH = ("YEAR", "QUARTER", "MONTH")
 
 
+def constraint_sql(self, expression: exp.Constraint) -> str:
+    this = self.sql(expression, "this")
+
+    # remove unsupported properties
+    expression.expressions.remove(expression.find(exp.Properties))
```

Comment:
i don't understand what this is doing, it doesn't look correct

Reply:
Hive / Spark support some constraints but not all constraints other dialects that might emit.

This code intends to filter out the non-supported constraints, so this code does a find and removes them.

Reply:
what constraints are being removed though? just all properties?, i think you can just do

for prop in expression.find_all(exp.Properties):
    prop.pop()

Reply:
Is `remove` bad? It worked better than iterating over a collection and popping from the same collection which appeared to create a race condition.
Is the concern that properties are removed or how the properties are being removed?

Reply:
this is not resolved, you didn't answer the question about why this is being removed.

Reply:
it seems too broad, and it should use pop, and not remove